src/mysql-util/src/replication.rs

```diff
@@ -50,10 +58,236 @@ pub async fn ensure_gtid_consistency(conn: &mut Conn) -> Result<(), MySqlError>
     Ok(())
 }
 
+// TODO: This only applies if connecting to a replica, to ensure that the replica preserves
+// all transactions in the order they were committed on the primary which implies
+// monotonically increasing transaction ids
 pub async fn ensure_replication_commit_order(conn: &mut Conn) -> Result<(), MySqlError> {
     // This system variable was renamed between MySQL 5.7 and 8.0
     match verify_sys_setting(conn, "replica_preserve_commit_order", "1").await {
         Ok(_) => Ok(()),
         Err(_) => verify_sys_setting(conn, "slave_preserve_commit_order", "1").await,
     }
 }
+
+#[derive(Debug, Clone, Deserialize, Serialize, Hash, PartialEq, Eq)]
+pub struct GtidInterval {
+    pub start: u64,
+    pub end: Option<u64>,
```

Comment:
might want to document if this is an open-end, or if its means a single point if its `None`

Reply:
It seems like its the later, which to me suggests that making `end` not optional is better (just copy the u64 from start to end)